modules/ROOT/pages/field-level-capabilities.adoc

```diff
@@ -0,0 +1,256 @@
+= Field Level Capabilities
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+There are several features for Java SDK that support field level capabilities
+for value providers and sample data. Using these features enables you to define
+value providers for parameter fields, or to use parameter fields as acting
+parameters for either a value provider or sample data.
+
+These features are available for Mule 4.4.0, however, connectors can use
+the features without upgrading their minimum Mule version.
+
+== Defining a Value Provider for a Part of a Parameter
+
+You can declare a value provider for a parameter field by using the
+annotation *@FieldValues*.
+
+The *@FieldValues* annotation requires:
+
+* *value*
++
+Provides values for a field.
+* *targetSelectors*
++
+Describes which part of a parameter contains the values that are given by the value provider.
+It is defined through the DataWeave syntax to describe the selector through
+nested fields, XML attributes, or fields that contain spaces in their names.
+
+The following is an example of an operation that includes values provided for
+a parameter named `body`, which is a JSON input stream:
+
+----
+public class ChatOperations {
+ public void sendChatMessage(@Connection ChatConnection chatConnection,
+       @FieldValues(value = WorkspacesValueProvider.class, targetSelectors = "routingInfo.workspace")
+       @TypeResolver(ChatMessageResolver.class) @Content InputStream body) {
+    // Do message sending
+  }
+}
+----
+
+The `body` parameter has a field named *routingInfo*, which has a field named
+`workspace`. *WorkspacesValueProvider* is a value provider that uses the
+connection and retrieves possible workspaces.
+
+== Defining a Value Provider for Several Parts of a Parameter
+
+Parameters can have more than one part where values are provided.
+The *@FieldValues* annotation is a repeatable annotation, so parameters can be
+annotated multiple times.
```

Comment:
How is the audience involved? 
We begin the peer topics by giving advice about what "you" do to accomplish each particular definition, and why; try to follow parallel patterns in structuring the information.

Reply:
Here's my lame attempt...:
```suggestion
Because parameters sometimes require you to provide a value in more than one location, use *@FieldValues* to annotate parameters multiple times.
```

Reply:
Super helpful, thanks for this tip!